app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,62 @@
+class RecipesController < ApplicationController
+  load_and_authorize_resource except: :public_recipes
+  before_action :set_recipe, only: %i[show destroy]
+
+  # GET /recipes
+  def index
+    @recipes = current_user.recipes.all
+  end
+
+  # GET /public_recipes
+  def public_recipes
+    @recipes = Recipe.includes(:user, :recipe_foods).where({ public: true }).order(created_at: :desc)
+  end
```

Comment:
- [ ] You should avoid making SQL queries from the controller. Your controllers should only redirect data that they get from models. You should always make SQL queries from the model.